### PR TITLE
 Add swap() tests for Pool 4 

### PR DESCRIPTION
**Pull request by uri-99**
This PR adds the swap tests corresponding to the Pool 4, resolving issue #165 

It currently has a bugged swap case, fixable by PR #204

### CODE REVIEW

File: crates/yas_core/src/libraries/tick_bitmap.cairo [hunk 1]
```diff
@@ -152,12 +152,12 @@ mod TickBitmap {
             tick / divisor
         } else if tick >= negative_lower_bound { //tick: [-256, -1]
             one_negative
-        } else {
-            let mut ans = IntegerTrait::<i32>::new((tick.mag / divisor.mag) - 1, true);
-            if (mod_i32(tick, divisor) == zero) {
-                ans = ans + IntegerTrait::<i32>::new(1, false);
+        } else { //[-887272, -257]
```
Comment (dubzn) [inline crates/yas_core/src/libraries/tick_bitmap.cairo:155]: remove `//[-887272, -257]`

File: crates/yas_core/src/tests/utils/pool_4.cairo [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+    //     amount_1_before: 632455532033675867,
+    //     amount_1_delta: IntegerTrait::<i256>::new(634358607857247611, false),
+    //     execution_price: 15893312440173387730977230182,
+    //     fee_growth_global_0_X128_delta: 0,
+    //     fee_growth_global_1_X128_delta: 323791572837503501799197590655727195,
+    //     pool_price_after: 50108289675009586237282760313,
+    //     pool_price_before: 25054144837504793118641380156,
+    //     tick_after: IntegerTrait::<i32>::new(9164, true),
+    //     tick_before: IntegerTrait::<i32>::new(23028, true),
+    // },
```
Comment (dubzn) [inline crates/yas_core/src/tests/utils/pool_4.cairo:332]: remove this comments please

**Review by dubzn — CHANGES_REQUESTED** [review]
Remove some innecesary code comments :sparkle:

File: scripts/utility/calc_x96_diff.py
Comment (dubzn) [inline scripts/utility/calc_x96_diff.py:1]: why is this file here? :thinking:

File: crates/yas_core/src/libraries/tick_bitmap.cairo (hunk 1 above)
Comment (uri-99) [reply crates/yas_core/src/libraries/tick_bitmap.cairo]: these comments are for a better understanding of the code. They add legibility

File: scripts/utility/calc_x96_diff.py
Comment (uri-99) [reply scripts/utility/calc_x96_diff.py:1]: why is it not on test/swap? did we decide to delete this file? maybe we should

Comment (rcatalan98) [reply scripts/utility/calc_x96_diff.py:1]: yes we should remove it. It's not longer necessary

File: crates/yas_core/src/libraries/tick_bitmap.cairo (hunk 1 above)
Comment (rcatalan98) [reply crates/yas_core/src/libraries/tick_bitmap.cairo]: you could add some additional text explaining what it is to add more context

**Review by dubzn — APPROVED** [review]

**Review by rcatalan98 — APPROVED** [review]
